### PR TITLE
add max-active-tasks-per-worker configuration

### PR DESCRIPTION
**Pull request by pivotal-jamie-klassen**
since its introduction in concourse/concourse#4118

### CODE REVIEW

File: jobs/web/spec [hunk 1]
```diff
@@ -734,9 +734,20 @@ properties:
...
     default: "volume-locality"
 
+  max_active_tasks_per_worker:
+    env: CONCOURSE_MAX_ACTIVE_TASKS_PER_WORKER
+    description: |
+      Maximum allowed number of active build tasks per worker.
+
+      Has effect only when used with "limit-active-tasks" placement strategy.
+
+      0 means no limit.
```
Comment (nader-ziada) [inline jobs/web/spec]: Can we mention that this is an experimental feature

File: jobs/web/spec [hunk 2]
```diff
@@ -734,9 +734,20 @@ properties:
     description: |
       Method by which a worker is selected during container placement.
 
-      Options are "volume-locality", "random", and "least-build-containers".
+      Options are "volume-locality", "random", "fewest-build-containers", and
+      "limit-active-tasks".
```
Comment (ddadlani) [inline jobs/web/spec]: Can we say something like:

```
Supported options are "volume-locality", "random" and "least-build-containers".
Experimental option: "limit-active-tasks"
```

File: jobs/web/spec [hunk 3]
```diff
@@ -734,9 +734,20 @@ properties:
     description: |
       Method by which a worker is selected during container placement.
 
-      Options are "volume-locality", "random", and "least-build-containers".
+      Supported options are "volume-locality", "random" and "least-build-containers".
```
Comment (deniseyu) [inline jobs/web/spec]: should be `fewest-build-containers`

**Comment by deniseyu** [comment]
Other than documentation tweaks LGTM 👍

**Comment by ddadlani** [comment]
hold off until concourse/concourse#4161 goes through. The migration number is invalid, we don't want that to go on to prod. Right now because the bosh release doesn't match the code, this feature hasn't made it to prod

File: jobs/web/spec (hunk 3 above)
Comment (ddadlani) [reply jobs/web/spec]: thanks for pointing that out. done!

**Pull request merged**